### ISSUE TITLE
Improve out of cards error.

### ISSUE THREAD

**Issue by Lattyware** [Lattyware/massivedecks#104]
Currently this just kills the game. It would be nice to allow the addition of another deck to solve the problem and continue.

**Comment by Follpvosten** [comment]
Alternatively, there could also be a simple switch allowing the cards to repeat like they did previously (basically shuffling the deck and starting anew). Would be a very nice quick fix because this out-of-cards state is sadly killing our games all the time :)

**Comment by Follpvosten** [comment]
First of all, thanks for taking the time for such a thorough answer!
Yes, this may well be a bug. If it really is, I'll create a separate issue for it.
This is our general setup, to give you some context:

* We are four friends, and we create all of our cards ourselves. Because creating so many white cards is hard, we set the hand size to 8. We had 234 white cards and 64 black cards. (I am now realizing that the ratio there is way off, so cards would definitely have repeated with the previous version of massivedecks.)
* Our rules:
  * Hand Size: 8
  * Awesome Point Limit: Off
  * Rando Cardrissian: Off
  * Packing Heat: On
  * Rebooting the universe: Off
  * Comedy Writer: Off
* Time limits are all turned off, Finishing Time Limit is left at the default.
* Privacy settings are left at the default.
* No new players joined any of our games.

Now when all of the white cards have been dealt, but there are still black cards left:

Expected behavior: The white cards start repeating and the game continues normally.
Actual behavior: As soon as we play our cards in the (now sadly final) round, there's an error (which you can't really read because it's rendered behind the cards) telling us that the game ran out of cards and we should try adding more decks.

As to your last question, I think there was just one blank slot on the black card.

I hope this is enough information to at least determine if this is a real bug or intended behavior!

**Comment by Lattyware** [comment]
@Follpvosten Yeah, that definitely sounds like a bug. With that many cards and that few players, there is no way you should be running out of cards like that. I'll take a look at why it isn't reshuffling properly when I get a chance.

**Comment by Follpvosten** [comment]
Thanks a lot for your work!

**Comment by lhvy** [comment]
> You only get this error when there are not enough cards in the game for everyone in the game to have a full hand of cards (i.e: we go to shuffle the discarded cards and there are none to shuffle), and so should only happen when new players join or you have black cards with a lot of blank spaces so a lot of cards end up "in use".

Maybe when this happens, the game could pause and the host could receive a pop-up that tells them they are out of cards and gives them two options:
1. Add a new deck/decks until at required cards
2. Fill in the missing cards with random duplicate cards from in-use decks
